packages/gatsby/src/redux/actions.js

```diff
@@ -596,6 +596,9 @@ actions.createNode = (
           actions.deleteNode({ node: getNode(n) })
         )
       }
+      if (deleteAction) {
+        deleteAction.push(actions.deleteNode({ node: oldNode }))
+      }
```

Comment:
can we pass `oldNode` to `CREATE_NODE` action like you had in previous version and move deleting node to `loki` specific code? It feels that we shouldn't cover lack of upsert functionality here